daphne/src/roles_test.rs

```diff
@@ -43,10 +41,131 @@ macro_rules! get_reports {
     }};
 }
 
-impl MockAggregator {
+struct Test {
+    now: Time,
+    leader: MockAggregator,
+    helper: MockAggregator,
+    #[allow(dead_code)] // TODO(issue #100) Remove
+    collector_hpke_receiver_config: HpkeReceiverConfig,
+    #[allow(dead_code)] // TODO(issue #100) Remove
+    collector_token: BearerToken,
+    time_interval_task_id: Id,
+    #[allow(dead_code)] // TODO(issue #100) Remove
+    fixed_size_task_id: Id,
+}
+
+impl Test {
+    fn new() -> Self {
+        let now = SystemTime::now()
+            .duration_since(SystemTime::UNIX_EPOCH)
+            .unwrap()
+            .as_secs();
+        let mut rng = thread_rng();
+
+        // Global config. In a real deployment, the Leader and Helper may make different choices
+        // here.
+        let global_config = DapGlobalConfig {
+            max_batch_duration: 360000,
+            min_batch_interval_start: 259200,
+            max_batch_interval_end: 259200,
+            supported_hpke_kems: vec![HpkeKemId::X25519HkdfSha256],
+        };
+
+        // Task Parameters that the Leader and Helper must agree on.
+        let vdaf_config = VdafConfig::Prio3(Prio3Config::Count);
+        let leader_url = Url::parse("https://leader.biz/v01/").unwrap();
+        let helper_url = Url::parse("http://helper.com:8788/v01/").unwrap();
+        let time_precision = 3600;
+        let version = DapVersion::Draft01;
+        let collector_hpke_receiver_config =
+            HpkeReceiverConfig::gen(rng.gen(), HpkeKemId::X25519HkdfSha256);
+
+        // Create the task list.
+        let time_interval_task_id = Id(rng.gen());
```

Comment:
These are now randomly generated, as opposed to being fixed in a JSON test file, right? Should they be deterministically derived to help with test reproducibility?

Reply:
How would this help with reproducibility? The upside of generating it randomly is we avoid writing test code that hardcodes this. (Hardcoding saves us the pain of having to change the value in multiple places when we need to update it. This is already a problem for daphne_worker_test.)

Reply:
Say, for example, a test fails but only for a specific config ID. It'd be a shame if we lost that into the abyss because we never write down the config ID anywhere. (Not a blocking comment -- more of a question.)

Reply:
Yeah I get what you're saying. Of course it's possible to fix constants whenever we want a regression test for a behavior that dependent constant. But for the "baseline" test I don't think it's necessary.